envoy/secret/secret_provider.h

```diff
@@ -78,5 +78,15 @@ using GenericSecretConfigProvider =
     SecretProvider<envoy::extensions::transport_sockets::tls::v3::GenericSecret>;
 using GenericSecretConfigProviderSharedPtr = std::shared_ptr<GenericSecretConfigProvider>;
 
+struct TlsCertificateConfigProviderWithName {
```

Comment:
I don't think these types need to be defined in this interface, as they aren't used here. I think they could be defined in context_impl.h, probably near where the functions that return them are declared.